src/test/isolation2/input/resgroup/resgroup_startup_memory.source

```diff
@@ -0,0 +1,109 @@
+-- start_ignore
+drop table if exists t1;
+drop role if exists test1;
+drop role if exists test2;
+drop resource group rg1;
+drop resource group rg2;
+-- end_ignore
+
+create extension if not exists gp_inject_fault;
+
+create or replace function resGroupPalloc(float) returns int as '@abs_builddir@/../regress/regress@DLSUFFIX@', 'resGroupPalloc' language C reads sql data;
+create or replace function hold_memory_by_percent_on_qe(float) returns int as $$
+select resGroupPalloc($1) from gp_dist_random('gp_id')
+$$ language sql;
+
+create or replace function repeatPalloc(int, int) returns int as
+'@abs_builddir@/../regress/regress@DLSUFFIX@', 'repeatPalloc'
+    language C reads sql data;
+
+-- In the next test, we need to be able to allocate 100% of the available
+-- memory. Therefore, we set the memory_limit for the built-in groups to 0.
+-- memory_spill_ratio must also be set to 0.
+alter resource group admin_group set memory_spill_ratio 0;
+alter resource group default_group set memory_spill_ratio 0;
+alter resource group admin_group set memory_limit 0;
+alter resource group default_group set memory_limit 0;
+
+create resource group rg1 with (cpu_rate_limit=20, memory_limit=15, memory_shared_quota=100);
+create resource group rg2 with (cpu_rate_limit=20, memory_limit=85, memory_shared_quota=100);
+
+create role test1 with resource group rg1;
+create role test2 with resource group rg2;
+
+1: set role test1;
+1: create table t1 (a int) distributed randomly;
+-- We need to have a random distribution so that each select in the join is
+-- performed in a separate slice.
+-- Add exactly one line to each segment.
+0U: insert into t1 values (1);
+1U: insert into t1 values (2);
+2U: insert into t1 values (3);
+0Uq:
+1Uq:
+2Uq:
+
+-- Force nested loop join to avoid unnecessary memory allocations.
+1: set enable_nestloop = on;
+1: set enable_hashjoin = off;
+
+set role test2;
+begin;
+
+-- Allocate all the memory in the second resource group.
+select hold_memory_by_percent_on_qe(1);
+-- Wait for all backends to run before executing the query. This is necessary
+-- to avoid executing hold_memory before all backends are up.
+select gp_inject_fault_infinite('executor_pre_tuple_processed', 'suspend', dbid)
+    from gp_segment_configuration where role = 'p' and content > -1;
+-- The first resource group has 682 * 0.15 = 102 MB of memory available.
+-- 6 slices consume 12 * 6 = 72 MB, plus one slice explicitly allocates 40 MB of
```

Comment:
Can we ask OS how much memory a slice consumes?

Reply:
It seems that there is no known and well-documented way to get the exact amount of private memory of a process. Therefore, this is a topic for a separate study.